multiversion/src/main/scala/multiversion/outputs/ResolutionIndex.scala

```diff
@@ -169,36 +169,47 @@ object ResolutionIndex {
     def lessThan(v1: Version, v2: Version): Boolean =
       (!hasOverride(v1) && hasOverride(v2)) || (v1 < v2)
     // The "winners" are the highest selected versions
-    val winners = mutable.Set.empty[Version]
-    val versions = vers.map(Version(_))
-    versions.foreach { version =>
-      val isCompatible = winners.exists { winner =>
-        if (isCompat(version.repr, winner.repr, compat)) {
-          if (lessThan(winner, version) && !isUnstable(version)) {
-            winners.remove(winner)
-            winners.add(version)
-          }
-          true
-        } else {
-          false
+    val winners = mutable.Set.empty[(Version, Boolean)]
+    verForces.foreach {
+      case (v, force) =>
+        val version = Version(v)
+        val isCompatible = winners.exists {
+          case (winner, wforce) =>
+            if (isCompat(version.repr, winner.repr, compat)) {
+              if (
+                lessThan(winner, version)
+                && (force || !wforce)
+                && !isUnstable(version)
```

Comment:
Do I understand correctly that with these changes, the forced version will be kept if and only the current winner is less than the forced version? Shouldn't we keep the greatest of the forced versions, or the greatest of the unforced version if none are forced (within groups of compatible versions)?

I'm not entirely sure how to re-formulate what I'm asking more clearly. Should the condition be

```scala
(
     lessThan(winner, version) // select `version` if it's greater than the current winner
  || (force && !wforce)        // or if `version` is forced, but the winner was not forced
) 
&& !isUnstable(version) // unstable versions never win
```

?

Reply:
I think you're right. I need to write more tests to capture those cases.

Reply:
485936d3706574cc7be1604b08d61e48c2e1b2f8